cmd/net.go

```diff
@@ -186,6 +189,114 @@ func checkPortAvailability(port string) (err error) {
 	return nil
 }
 
+// extractHostPort - extracts host/port from many address formats
+// such as, ":9000", "localhost:9000", "http://localhost:9000/"
+func extractHostPort(hostAddr string) (string, string, error) {
+	var addr, scheme string
+
+	// Parse address to extract host and scheme field
+	u, err := url.Parse(hostAddr)
+	if err != nil {
+		// Ignore scheme not present error
+		if !strings.Contains(err.Error(), "missing protocol scheme") {
+			return "", "", err
+		}
+	} else {
+		addr = u.Host
+		scheme = u.Scheme
+	}
+
+	// Use the given parameter again if url.Parse()
+	// didn't return any useful result.
+	if addr == "" {
+		addr = hostAddr
```

Comment:
```
root@escape:~# minio gateway --address :80 s3 localhost
FATA[0000] Unable to compare server and endpoint addresses.  cause=unable to guess port from scheme source=[gateway-main.go:197:gatewayMain()]
root@escape:~#
```

is fixed by:
```
--- a/cmd/net.go
+++ b/cmd/net.go
@@ -210,6 +210,7 @@ func extractHostPort(hostAddr string) (string, string, error) {
        // didn't return any useful result.
        if addr == "" {
                addr = hostAddr
+               scheme = "http"
        }
 
        // At this point, addr can be one of the following form:

```

I have half a mind to ask if this PR is really worth the code added. But the code is well written and readable and hence won't ask ;-)